src/model/OutputData.ts

```diff
@@ -0,0 +1,169 @@
+				? undefined
+				: ({
+						s: hexToBytes(sig.dleq.s),
+						e: hexToBytes(sig.dleq.e),
+						r: this.blindingFactor
+				  } as DLEQ);
+		const blindSignature = {
+			id: sig.id,
+			amount: sig.amount,
+			C_: pointFromHex(sig.C_),
+			dleq: dleq
+		};
+		const A = pointFromHex(keyset.keys[this.blindedMessage.amount]);
+		const proof = constructProofFromPromise(blindSignature, this.blindingFactor, this.secret, A);
+		const serializedProof = {
+			...serializeProof(proof),
+			...(dleq && {
+				dleqValid: verifyDLEQProof_reblind(this.secret, dleq, proof.C, A)
+			}),
+			...(dleq && {
+				dleq: {
+					s: bytesToHex(dleq.s),
+					e: bytesToHex(dleq.e),
+					r: numberToHexPadded64(dleq.r ?? BigInt(0))
+				} as SerializedDLEQ
+			})
+		} as Proof;
+		return serializedProof;
+	}
+
+	static createP2PKData(
+		p2pk: { pubkey: string; locktime?: number; refundKeys?: Array<string> },
+		amount: number,
+		keyset: MintKeys,
+		customSplit?: Array<number>
+	) {
+		const amounts = splitAmount(amount, keyset.keys, customSplit);
+		return amounts.map((a) =>
+			this.createSingleP2PKData(p2pk.pubkey, a, keyset.id, p2pk.locktime, p2pk.refundKeys)
+		);
+	}
+
+	static createSingleP2PKData(
+		pubkey: string,
+		amount: number,
+		keysetId: string,
+		locktime?: number,
+		refundKeys?: Array<string>
+	) {
+		const newSecret: [string, { nonce: string; data: string; tags: Array<any> }] = [
+			'P2PK',
+			{
+				nonce: bytesToHex(randomBytes(32)),
+				data: pubkey,
+				tags: []
+			}
+		];
+		if (locktime) {
+			newSecret[1].tags.push(['locktime', locktime]);
+		}
+		if (refundKeys) {
+			newSecret[1].tags.push(['refund', refundKeys]);
+		}
+		const parsed = JSON.stringify(newSecret);
+		const secretBytes = new TextEncoder().encode(parsed);
+		const { r, B_ } = blindMessage(secretBytes);
+		return new OutputData(
+			new BlindedMessage(amount, B_, keysetId).getSerializedBlindedMessage(),
+			r,
+			secretBytes
+		);
+	}
+
+	static createRandomData(amount: number, keyset: MintKeys, customSplit?: Array<number>) {
+		const amounts = splitAmount(amount, keyset.keys, customSplit);
+		return amounts.map((a) => this.createSingleRandomData(a, keyset.id));
+	}
+
+	static createSingleRandomData(amount: number, keysetId: string) {
+		const randomHex = bytesToHex(randomBytes(32));
+		const secretBytes = new TextEncoder().encode(randomHex);
+		const { r, B_ } = blindMessage(secretBytes);
+		return new OutputData(
+			new BlindedMessage(amount, B_, keysetId).getSerializedBlindedMessage(),
+			r,
+			secretBytes
+		);
+	}
+
+	static createDeterministicData(
+		amount: number,
+		seed: Uint8Array,
+		counter: number,
+		keyset: MintKeys,
+		customSplit?: Array<number>
+	): Array<OutputData> {
+		const amounts = splitAmount(amount, keyset.keys, customSplit);
+		const data: Array<OutputData> = [];
+		for (let i = 0; i < amounts.length; i++) {
+			data.push(this.createSingleDeterministicData(amount, seed, counter + i, keyset.id));
+		}
+		return data;
```

Comment:
```suggestion
		return amounts.map((_, i) =>
			this.createSingleDeterministicData(amount, seed, counter + i, keyset.id)
		);
```

or should it be
```
return amounts.map((a, i) =>
	this.createSingleDeterministicData(a, seed, counter + i, keyset.id)
);
```
?

Reply:
Yes, good point. Fixed